### PR TITLE
Refact extract try get constructor

### PR DESCRIPTION
**Pull request by omsmith**
On top of https://github.com/Brightspace/D2L.CodeStyle/pull/442

### CODE REVIEW

**Comment by j3parker** [comment]
Your commit is actually in the middle of #442's commits. Rebase?

**Comment by j3parker** [comment]
(Probably doesn't actually matter because of squash)

**Comment by omsmith** [comment]
Hmm, yeah. Rebase, but I think GitHub orders by date or something?

File: src/D2L.CodeStyle.Analyzers/ApiUsage/DependencyInjection/DependencyRegistrationsAnalyzer.cs [hunk 1]
```diff
@@ -208,13 +208,22 @@ ITypeSymbol type
 			return ImmutableArray.Create( registration.DependencyType );
 		}
 
-		private bool TryGetDependenciesFromConstructor( ITypeSymbol type, out ImmutableArray<ITypeSymbol> dependencies  ) {
-			var ctor = type.GetMembers()
+		private bool TryGetInjectableConstructor( ITypeSymbol type, out IMethodSymbol injectableConstructor ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/ApiUsage/DependencyInjection/DependencyRegistrationsAnalyzer.cs]: Dunno if there's a reason everything can't be `INamedTypeSymbol` (don't expect so), but it has a `.Constructors` if we did

**Comment by j3parker** [comment]
Oh, ya, I only looked at GitHub. Grumble grumble.

**Pull request merged**